src/Data/Nat/Primality.agda

```diff
@@ -294,6 +294,17 @@ prime⇒rough (prime pr) = pr
 rough∧∣⇒prime : .{{NonTrivial p}} → p Rough n → p ∣ n → Prime p
 rough∧∣⇒prime r p∣n = prime (rough∧∣⇒rough r p∣n)
 
+-- If a number n is m-rough, and m * m > n, then n must be prime.
+rough∧>square⇒prime : .{{NonTrivial n}} → m Rough n → m * m > n → Prime n
+rough∧>square⇒prime rough m*m>n = prime ¬composite
```

Comment:
Re: name of this lemma: `rough∧<square⇒prime`? 
In any case, probably room for a (slightly) improved name?